File: app/src/main/java/com/verbatoria/di/schedule/ScheduleComponent.java
package com.verbatoria.di.schedule;

import com.verbatoria.ui.schedule.view.ScheduleActivity;

import dagger.Subcomponent;

/**
 * Компонент Даггера для расписания
 *
 * @author nikitaremnev
 */
@Subcomponent(modules = {ScheduleModule.class})
@ScheduleScope
public interface ScheduleComponent {

    void inject(ScheduleActivity scheduleActivity);

}

File: app/src/main/java/com/verbatoria/di/schedule/ScheduleComponent.kt
package com.verbatoria.di.schedule

import com.verbatoria.di.BaseInjector
import com.verbatoria.ui.schedule.ScheduleActivity
import dagger.Subcomponent

/**
 * @author n.remnev
 */

@Subcomponent(modules = [ScheduleModule::class])
interface ScheduleComponent : BaseInjector<ScheduleActivity> {

    @Subcomponent.Builder
    interface Builder {

        fun build(): ScheduleComponent

    }

}